src/core/ts/background-scripts/background.js/telemetry/Telemetry.ts

```diff
@@ -5,78 +5,149 @@
 import Glean from "@mozilla/glean/webext";
 import { custom } from "./generated/pings";
 import { config } from "../../../config";
+import {
+  modelLoadTime,
+  translationTime,
+  wordsPerSecond,
+} from "./generated/performance";
+import { fromLang, toLang } from "./generated/metadata";
+import {
+  displayed,
+  changeLang,
+  closed,
+  neverTranslateSite,
+  translate,
+  neverTranslateLang,
+  notNow,
+} from "./generated/infobar";
 
+/**
+ * This class contains general telemetry initialization and helper code and synchronous telemetry-recording functions.
+ *
+ * Synchronous methods here is important, since it is the only way to guarantee that multiple Glean API calls are
+ * executed sequentially and not interleaved with other asynchronous Telemetry recording.
+ * For more information, see: https://github.com/mozilla-extensions/bergamot-browser-extension/pull/76#discussion_r602128568
+ */
 export class Telemetry {
-  private static _instance = null;
-  private _metricsToSubmit;
-
   constructor() {
     const appId = config.telemetryAppId;
     Glean.initialize(appId, true, {
```

Comment:
Just to add some extra context here: This will create what looks like a "new" user every time the extension starts up, and make it impossible to calculate accurate DAU.